src/utils/miscellaneous.ts

```diff
@@ -210,3 +210,22 @@ export function notEmpty<TValue>(value: TValue | null | undefined): value is TVa
 }
 
 export const isNonZeroNumber = (value?: string | number): boolean => !!value && !!+value
+
+export interface TimeoutParams<T> {
+  time?: number
+  result?: T
+  timeoutErrorMsg?: string
+}
+
+export function timeout<T>(params: TimeoutParams<T>): Promise<T> {
```

Comment:
Types are not exactly correct. You intentionally throw (return rejected Promise) for some `T`. Typescript can detect that if you do an overload type, for example